### PR TITLE
set parent operation right away in non lazy IOperation implementation

### PR DESCRIPTION
**Pull request by heejaechang**
we have 2 implementation of IOperation. lazy and non-lazy one. we used to set Parent operation lazily in both case, but now we only set it lazily in lazy implementation. another implementation, we set it right away.

### CODE REVIEW

**Comment by heejaechang** [comment]
@AlekseyTs @dotnet/roslyn-analysis can you take a look? basically changes are removing xxxImpl indirection in base type and each implementation handles setting parent differently.

**Comment by heejaechang** [comment]
@jinujoseph what is policy for sign off for dataflow feature branch? 2 compiler dev sign off?

**Comment by jinujoseph** [comment]
yes , 2 sign off from compiler

**Comment by AlekseyTs** [comment]
@cston Please review

File: src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs [hunk 1]
```diff
@@ -50,10 +49,10 @@ internal sealed partial class AddressOfExpression : BaseAddressOfExpression, IAd
         public AddressOfExpression(IOperation reference, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
             base(semanticModel, syntax, type, constantValue, isImplicit)
         {
-            ReferenceImpl = reference;
+            Reference = Operation.SetParentOperation(reference, this);
```
Comment (cston) [inline src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: Is `Operation.` qualifier necessary for these calls?

Comment (heejaechang) [reply src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs]: I can remove those.

**Comment by heejaechang** [comment]
updated PR. removed unnecessary Operaton from Operation.SetParentOperation call

**Comment by heejaechang** [comment]
@333fred @AlekseyTs can you take a look again?

**Pull request merged**